src/_blog/libp2p-and-Ethereum.md

```diff
@@ -0,0 +1,280 @@
+
+In this talk David gave an overview of libp2p and gave a demo. He ran the EVM in a browser and modified a go-ethereum/Geth node to use libp2p. The Geth node running libp2p connected to the Ethereum mainnet, and the EVM nodes in the browsers (also running libp2p) were able to fetch blocks from the Geth node.
+At the time, this was a big breakthrough and a sign of things to come.
+
+Following this presentation, Protocol Labs and Ethereum [continued working together](https://github.com/ipfs/notes/issues/173) to see how it would be possible to integrate libp2p.
+
+<div class="container" style="display:flex; column-gap:10px;">
+    <figure>
+        <img src="https://i.imgur.com/hfy2O8S.png"/>
+    </figure>
+</div>
+
+Conversations continued into Devcon 3 in 2017.
+There Felix gave a [talk on devp2p](https://archive.devcon.org/archive/watch/3/evolving-devp2p/?tab=YouTube) which mentioned integration options with libp2p.
+Following the event, he published the "[Ethereum ♥ libp2p](https://twurst.com/articles/eth-loves-libp2p.html)" article that expanded on what a libp2p integration could look like:
+
+*"I've been thinking about ways to integrate libp2p for quite a while...My initial goal was to join their DHT, which turned out to be impractical. There's a much easier thing we can do though: use libp2p's transports and multiplexing algorithm. This saves us a bunch of code and gives us a few new protocols...that we can try out without too much work...The ultimate goal is to adopt libp2p's transport protocols as the way Ethereum nodes communicate.*"
+
+This showcased the desire by both communities to leverage the benefits of libp2p.
+
+##  Designing libp2p into the Beacon Chain (2018-2019) :writing_hand: 
+The above discussions were about integrating libp2p into the Ethereum mainnet, either replacing or augmenting devp2p.
+
+In 2018, the Ethereum community put major focus on how to scale the network. As the year progressed, the plan became to deliver both scalability and proof-of-stake as a part of "Ethereum 2.0" (for more information, see: [State of the Ethereum Protocol #1](https://media.consensys.net/state-of-ethereum-protocol-1-d3211dd0f6)).
+Both solutions were to be developed on a separate chain from the mainnet, called the [Beacon Chain](https://ethereum.org/en/upgrades/beacon-chain/).
+
+At this time, the Ethereum 2.0 community began evaluating the networking requirements for the Beacon Chain. Designing a libp2p integration became a formal effort. Focus moved towards adding libp2p to the Beacon Chain instead of extending/replacing devp2p on the Ethereum mainnet.
+
+[Raul Kripalani](https://twitter.com/raulvk), then the libp2p team lead at Protocol Labs, joined in on many of the [early Eth 2.0 implementers calls](https://github.com/ethereum/eth2.0-pm/issues?q=is%3Aissue+libp2p+is%3Aclosed) and advocated for libp2p.
```

Comment:
A few missing story details worth highlighting:
- The Eth 2.0 client ecosystem involved 7-8 clients written in different languages.
- Several of those languages had no libp2p implementations and this was a hard blocker for libp2p adoption.
- This motivated the birth of several new libp2p implementations, e.g. nim-libp2p, py-libp2p, jvm-libp2p, etc.
- Some implementations were literally started by me to overcome the cold start problem.
- EF and PL cofunded those implementations.
- libp2p became the focus of EthBerlinZwei, awarding up to 20k? USD in bounties cofunded by PL, EF, ICF. (Can't remember).
- EthBerlinZwei was a phenomenal opportunity to conduct knowledge transfer and onboard Eth Core Devs onto libp2p, see: https://twitter.com/protolambda/status/1599438035594014720
- libp2p was invited to give several talks at Devcon4 (Prague) and Devcon5 (Osaka), as well as in follow-up events like Crosslink Taipei.
- There's a very cool tweet you can link to about the Eth 2.0 networking spec: https://twitter.com/jcksie/status/1156969492607590401 These were mad 2 weeks.

I can give you more links and help you capture more colour about this gradual marriage between communities throughout this week!

Reply:
Awesome, definitely want to highlight these. Thank you Raul! Added them in this commit: https://github.com/libp2p/blog/pull/32/commits/1ecec054cf35a2ebd4079c8828021dd044153857 if that makes it easier to review. [preview here](https://bafybeihgfw4jqxnxppqklc43i4sdaturq25bj7mb7s65xso7nl4jbgbrim.on.fleek.co/libp2p-and-ethereum/)